packages/block-library/src/navigation-link/index.php

```diff
@@ -266,6 +266,70 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 	return $html;
 }
 
+/**
+ * Registers a variation for a post type / taxonomy for the navigation link block.
+ *
+ * @since 6.5.0
+ * @deprecated 6.5.0 Use WP_Block_Type::get_variations / get_block_type_variations filter instead.
+ *
+ * TODO: After two WP versions (6.7), we can remove this.
```

Comment:
If this is here only to ensure compatibility with previous versions in the Gutenberg plugin, can we define these functions outside the "block-library" package or no?

Reply:
It's there for two reasons: 
- The functions have been in Gutenberg since November (but no core version)
- They provide backwards-compatibility when the Gutenberg plugin is used with WP versions < 6.5

Do you mean, I could move them into the `lib/compat/wordpress-6.5` directory?

Reply:
> Do you mean, I could move them into the lib/compat/wordpress-6.5 directory?

Yes, since these functions are "Gutenberg specific" and shouldn't "run" in Core. I was wondering if there's a way to keep the "navigation" block code "core only" and add the compat code from the compat folder. The reason is that the navigation block code will be copied as is into Core.

Reply:
@youknowriad I understand. 
I did that in dd6857cd725dbf3c3d94154388f4a1e3d4bd1513
I disabled the phpcs rules for these functions, since they were never in core and therefore imho shouldn't need a guarded clause. But if that thought was wrong, let me know. 
Tests are passing locally. Only two other tests related to `WP_Style_Engine_CSS_Rule_Gutenberg::set_at_rule` are failing locally - but let's see what the GitHub tests say.

Reply:
There's also some recent php unit test failures on trunk due to some recent core commit. So we should ignore the failures that are already in trunk.

Reply:
Maybe I'm missing something the functions are still in the navigation-link/index.php file?